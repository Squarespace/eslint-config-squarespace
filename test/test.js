const path = require('path');
const fs = require('fs');
const { CLIEngine } = require('eslint');

function lintFixtures(config, fixturePath) {
  const newConfig = {
    root: true,
    useEslintrc: false,
    baseConfig: config,
  };

  const cli = new CLIEngine(newConfig);
  const fixtures = fs
    .readdirSync(fixturePath)
    .filter((filename) => /\.js|\.ts$/.test(filename));

  return fixtures.map((fileName) => {
    const code = fs.readFileSync(path.join(fixturePath, fileName)).toString();
    const result = cli.executeOnText(code);
    return { fileName, result };
  });
}

describe('eslintConfig', function () {
  const rootPath = path.join(__dirname, '__fixtures__');
  const newConfig = Object.assign(require('../index.js'));
  const fixtures = lintFixtures(newConfig, rootPath);

  fixtures.forEach(({ fileName, result }) => {
    test(fileName, function () {
      const numOfErrors = result.errorCount + result.warningCount;
      const shouldReportErrors =
        fileName.indexOf('bad.js') === -1 && numOfErrors > 0;
      if (shouldReportErrors) {
        result.results.forEach(({ messages }) => {
          console.log('messages', messages);
        });
      }

      if (fileName.indexOf('bad.js') !== -1) {
        expect(numOfErrors).toBeGreaterThan(0);
      }
      expect(numOfErrors).toMatchSnapshot();
    });
  });
});