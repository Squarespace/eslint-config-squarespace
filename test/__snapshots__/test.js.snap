// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`eslintConfig vanilla.bad.js 1`] = `24`;

exports[`eslintConfig vanilla.good.js 1`] = `0`;
